src/codegen/encoding/mod.rs

```diff
@@ -469,13 +448,28 @@ fn load_sub_array(
             Box::new(Expression::Variable(
                 Loc::Codegen,
                 Type::Uint(32),
-                indexes[indexes.len() - i - 1],
+                indexes[dims.len() - i - 1],
             )),
         );
+
+        // Multidimensional arrays have an inconsistent indexing method. We should only load if the
+        // dimension is dynamic.
```

Comment:
A comment on why fixed size dimension do not need a load would be helpful. Also this comment would serve better as doc-comments to the function.